axelrod/strategies/punisher.py

```diff
@@ -139,8 +139,9 @@ class LevelPunisher(Player):
     }
 
     def strategy(self, opponent):
-        if len(opponent.history) <= 10: return D
-            else:
-                if float(len(opponent.history) - sum(opponent.history)) / len(opponent.history) > 0.2:
-                    return C
-                else: return D
+        if len(opponent.history) <= 10:
+            return D
+        elif float(len(opponent.history) - sum(opponent.history)) / len(opponent.history) > 0.2:
```

Comment:
Hey @souravsingh, `opponent.history` is a list of strings (`['C', 'D', 'C', ...]`) and not integers, which is why I think  `sum(opponent.history)` is causing some tests to fail. 

Here's an output of one of the failed tests:
```
======================================================================
ERROR: test_outcome_repeats (axelrod.tests.integration.test_matches.TestMatchOutcomes)
A test that if we repeat 3 matches with deterministic and well
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/tests/integration/test_matches.py", line 20, in test_outcome_repeats
    min_size=2, max_size=2),
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/hypothesis/core.py", line 524, in wrapped_test
    print_example=True, is_final=True
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/hypothesis/executors.py", line 58, in default_new_style_executor
    return function(data)
  File "/home/travis/virtualenv/python3.5.2/lib/python3.5/site-packages/hypothesis/core.py", line 111, in run
    return test(*args, **kwargs)
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/tests/integration/test_matches.py", line 27, in test_outcome_repeats
    self.assertEqual(matches[0].play(), matches[1].play())
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/match.py", line 121, in play
    self.players[0].play(self.players[1], self.noise)
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/player.py", line 160, in play
    s1, s2 = self.strategy(opponent), opponent.strategy(self)
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/strategies/punisher.py", line 145, in strategy
    elif float(len(opponent.history) - sum(opponent.history)) / len(opponent.history) > 0.2:
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

Since CoopSim represents 'C' and 'D' as 1 and 0, try using `opponent.cooperations` to get the number of cooperations instead.

Also, since Python 2 support has been dropped, it is unnecessary now to do the float cast as regular division will return a float regardless in Python 3.